src/EditorFeatures/Core/Implementation/CodeActions/CodeActionEditHandlerService.cs

```diff
@@ -123,7 +122,7 @@ internal class CodeActionEditHandlerService : ForegroundThreadAffinitizedObject,
                 {
                     // ConfigureAwait(true) so we come back to the same thread as 
                     // we do all application on the UI thread.                    
-                    if (!await document.HasAnyErrorsAsync(cancellationToken).ConfigureAwait(true))
+                    if (!document.HasAnyErrorsAsync(cancellationToken).Result)
```

Comment:
ickkkkkkkkkkkkkkkkkkkkkkkkkkkk.  I dn't think this is ok tbh.  need @jasonmalinowski @sharwell  here as well.

Reply:
Please note that this line is under 
`#if DEBUG && false`

Reply:
1. does this work properly with cancellation?  i.e. giving you cancellation-exception, and not aggregate exception?
2. does this work properly with other exceptions? 
3. we have asynchrony here, trying to push synchrony down to this level feels wrong.  feels like we should go *upwards* spreading asynchrony instead.

Reply:
oh.... phewwwwwwwwwwwwwwwwwwwwwww

Reply:
Oh yeah @CyrusNajmabadi is right: this should be .WaitAndGetResult().

Reply:
Yes, it's #if false but we should still make the code not be more wrong. :smile:

Reply:
```suggestion
                    if (!document.HasAnyErrorsAsync(cancellationToken).WaitAndGetResult(cancellationToken))
```